### PR TITLE
Use lookup_icon before choose_icon so hicolor theme searched.

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes https://github.com/elementary/icons/issues/808

### CODE REVIEW

File: libcore/IconInfo.vala [hunk 1]
```diff
@@ -90,9 +90,17 @@ public class Marlin.IconInfo : GLib.Object {
 
             var names = ((GLib.ThemedIcon) icon).get_names ();
             var theme = get_icon_theme ();
-            var gtkicon_info = theme.choose_icon_for_scale (names, size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
+
+            /* Looks in default theme and hicolor theme for primary icon name first */
+            var gtkicon_info = theme.lookup_icon_for_scale (names[0], size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
```
Comment (tintou) [inline libcore/IconInfo.vala]: Why not something like this?
```suggestion
            var gtkicon_info = theme.lookup_by_gicon_for_scale (icon, size, scale, Gtk.IconLookupFlags.FORCE_SIZE);
```

Comment (jeremypw) [reply libcore/IconInfo.vala]: It doesn't work, I am afraid.  It no longer finds hicolor icons in that form.  I do not know why exactly.

Comment (4jNsY6fCVqZv) [reply libcore/IconInfo.vala]: @jeremypw Does PR work in this way or what else does it need to be integrated into Files?

**Comment by 4jNsY6fCVqZv** [comment]
@jeremypw Does PR work in this way or what else does it need to be integrated into files?

**Comment by 4jNsY6fCVqZv** [comment]
Does PR work in this way or what else does it need to be integrated into Files?

**Comment by 4jNsY6fCVqZv** [comment]
@jeremypw Does PR work in this way or what else does it need to be integrated into Files?

**Comment by jeremypw** [comment]
@4jNsY6fCVqZv The PR worked the last time I checked it (some time ago), but I did not get it to work with @tintou 's suggested change.

**Comment by 4jNsY6fCVqZv** [comment]
@tintou Is your proposed change mandatory in order to merge this PR?

**Pull request merged**